docs-src/content/audio-input/_index.md

```diff
@@ -260,6 +223,319 @@ self.asrStream.result(completion: { (error) in
+this means it should be handled on a separate thread.
+
+{{< tabs >}}
+
+{{< tab "Go" "go" >}}
+var audioData []byte
+for haveAudioData {
+	audioData = getFromSource()
+
+	// Send the audio bytes
+	if err := stream.SendAudio(audioData); err == io.EOF {
+		// The stream has been closed, and a result is available.
+		break
+	}
+}
+
+// Be sure to close the stream to notify the server that no more
+// audio will be coming.
+err = stream.SendFinished()
+{{< /tab >}}
+
+{{< tab "Python" "python">}}
+while haveAudioData:
+    audioData = getFromSource()
+
+    # Send the audio bytes
+    if not stream.send_audio(audioData):
+        # The stream has been closed
+        break
+
+# Be sure to close the stream to notify the server that no more
+# audio will be coming.
+stream.send_finished()
+{{< /tab >}}
+
+{{< tab "C++" "c++" >}}
+while (haveAudioData) {
+    std::string audioData = getFromSource();
+
+    // Send the audio bytes
+    if (!stream.sendAudio(audioData)) {
+        // The stream has been closed
+        break;
+    }
+}
+
+// Be sure to close the stream to notify the server that no more
+// audio will be coming.
+stream.sendFinished();
+{{< /tab >}}
+
+{{< /tabs >}}
+
+### Transcribe Results
+Multiple results may be sent on the receiving half of the Transcribe
+stream during its lifetime. Some of these results are partial, which
+indicate that the result may change with further processing. Partial
+results are mainly useful for displaying progress to a user in a
+graphical user interface if one is being used. Non-partial results
+contain the final transcript that is of interest to the application.
+Note that multiple non-partial results may be accumlated by the
+application before the stream ends.
+
+Due to the bidirection nature of the Transcribe stream, results must
+be processed concurrently with sending audio data. This may be done
+on any thread, and it is often done on the main thread, as many
+applications don't wish to continue processing until the transcription
+is complete. However, it is not a strict requirement that results be
+processed on the main thread, and an application should adopt an
+architecture that makes the most sense for its use case.
+
+{{< tabs >}}
+
+{{< tab "Go" "go" >}}
+// Process results as they come.
+for {
+	result, err := stream.ReceiveResult()
+	if err == io.EOF {
+		// The stream has closed
+		return
+	} else if err != nil {
+		// Handle the error
+		return
+	}
+
+	// The transcript
+	fmt.Printf("text: %v\n", result.Text)
+
+	// The confidence the ASR system has in the given transcript,
+	// given as a value between 0 and 1.0, with 1.0 being the most
+	// confident in the result. Note that partial results won't
+	// have an accurate confidence score.
+	fmt.Printf("confidence: %v\n", result.Confidence)
+
+	// Boolean indicating whether the result is partial (i.e., may
+	// change with further processing).
+	fmt.Printf("partial: %v\n\n", result.IsPartial)
```

Comment:
I think it might be a more realistic and helpful example if we check for IsPartial and do something a bit different. This applies to all the language examples, but I'll use the go example to illustrate what I mean (I'm just writing this off the top of my head in github, so please forgive/correct any syntax or other errors)

```suggestion
	// Boolean indicating whether the result is partial (i.e., may
	// change with further processing).
	if result.IsPartial {
	    // Print the partial transcript
	    fmt.Printf("partial: %v...\n", result.Text)
	} else {
	    // The transcript
	    fmt.Printf("final: %v\n", result.Text)

  	    // The confidence the ASR system has in the given transcript,
	    // given as a value between 0 and 1.0, with 1.0 being the most
	    // confident in the result. Note that partial results don't
	    // have an accurate confidence score, so only printing confidence for the final result.
	    fmt.Printf("confidence: %v\n\n", result.Confidence)
        }

```